src/data.rs

```diff
@@ -48,12 +48,19 @@ pub async fn build_finder(ctx: &Context) -> Result<Nucleo<(u64, String)>> {
 
 pub struct Data {
     pub meme_finder: Mutex<Nucleo<(u64, String)>>,
+    pub http_client: HttpClient,
+    pub songbird: Arc<songbird::Songbird>,
 }
 
 impl Data {
-    pub async fn new(ctx: &Context) -> Result<Self> {
+    pub async fn new(ctx: &Context, songbird: Arc<songbird::Songbird>) -> Result<Self> {
         let meme_finder = Mutex::new(build_finder(ctx).await?);
+        let http_client = HttpClient::new();
```

Comment:
separated http_client is abundant, use ctx.http() instead

Reply:
We are using Songbird, which need reqwest client to create a request to search youtube